src/tools/miri/src/shims/unix/linux/sync.rs

```diff
@@ -35,6 +37,50 @@ pub fn futex<'tcx>(
     let futex_wake = this.eval_libc_i32("FUTEX_WAKE");
     let futex_wake_bitset = this.eval_libc_i32("FUTEX_WAKE_BITSET");
     let futex_realtime = this.eval_libc_i32("FUTEX_CLOCK_REALTIME");
+    let futex_lock_pi = this.eval_libc_i32("FUTEX_LOCK_PI");
+    let futex_unlock_pi = this.eval_libc_i32("FUTEX_UNLOCK_PI");
+    let futex_waiters = this.eval_libc_u32("FUTEX_WAITERS");
+    let futex_tid_mask = this.eval_libc_u32("FUTEX_TID_MASK");
+
+    // Ok(None) for EINVAL set, Ok(Some(None)) for no timeout (infinity), Ok(Some(Some(...))) for a timeout.
+    // Forgive me, I don't want to create an enum for this return value.
```

Comment:
I think this becomes cleaner if you don't pass in `dest`, and return an `InterpResult<'tcx, Result<Option<(...)>, IoError>>`. Then the caller should do `set_last_error`.